plugin-server/src/worker/ingestion/event-pipeline/1-populateTeamDataStep.ts

```diff
@@ -0,0 +1,31 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { PipelineEvent } from '../../../types'
+import { EventPipelineRunner, StepResult } from './runner'
+
+// TRICKY: team_id is optional on PipelineEvent but not for PluginEvent
```

Comment:
Nit: This comment is awkwardly placed - I was expecting description on what this method does, and instead it seems to describe something deep in this behavior.

Suggestion: Add a block describing what inputs may be coming in here, example.

```
/* 
This step populates event.team_id and event.ip if needed.

event.team_id might be unset in this particular condition.
*/
```